File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        queue<vector<TreeNode*> > nodesq;
        TreeNode *curr = root;
        if ( curr ){
            vector<TreeNode*> temp;
            temp.push_back(curr);
            nodesq.push(temp);
        }
        vector<vector<int> > res;
        while ( nodesq.size() > 0 ){
            vector<TreeNode*> temp = nodesq.front();
            nodesq.pop();
            vector<TreeNode*> test ;
            for ( int i = 0; i < temp.size(); i++ ){
                if ( temp[i]->left != NULL ) test.push_back(temp[i]->left);
                if ( temp[i]->right != NULL ) test.push_back(temp[i]->right);
            }
            if ( test.size() > 0) nodesq.push(test);
            vector<int> val ;
            for ( int j = 0; j < temp.size(); j++ ){
                val.push_back(temp[j]->val);
                
            }
            res.push_back(val);
        }
        return res;
    }
};

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
      Queue<List<TreeNode>> listQueue = new ArrayDeque<>();
    List<List<Integer>> retval = new ArrayList<>();
    if(root == null)
    {
      return Collections.EMPTY_LIST;
    }

    listQueue.add(List.of(root));

    retval.add(List.of(root.val));
    while(!listQueue.isEmpty())
    {
      List<TreeNode> currLevel = listQueue.poll();

      List<TreeNode> newList = new ArrayList<>();
      List<Integer> retList = new ArrayList<>();
      for( int i = 0; i < currLevel.size(); i++)
      {
        if(currLevel.get(i).left != null)
        {
          newList.add(currLevel.get(i).left);
          retList.add(currLevel.get(i).left.val);
        }
        if(currLevel.get(i).right != null)
        {
          newList.add(currLevel.get(i).right);
          retList.add(currLevel.get(i).right.val);
        }
      }
      if(!newList.isEmpty())
      {
        listQueue.add(newList);
        retval.add(retList);
      }

    }
    return retval;


    }
}